rt/frontend/jaxrs/src/main/java/org/apache/cxf/jaxrs/utils/InjectionUtils.java

```diff
@@ -539,21 +547,32 @@ private static RuntimeException createParamConversionException(ParameterType pTy
         }
         return ExceptionUtils.toBadRequestException(ex, null);
     }
-    public static <T> T createFromParameterHandler(String value,
-                                                    Class<T> pClass,
-                                                    Type genericType,
-                                                    Annotation[] anns,
-                                                    Message message) {
-        T result = null;
+    
+    public static <T> Optional<ParamConverter<T>> getParamConverter(Class<T> pClass,
+            Type genericType, Annotation[] anns, Message message) {
+        
         if (message != null) {
             ServerProviderFactory pf = ServerProviderFactory.getInstance(message);
             ParamConverter<T> pm = pf.createParameterHandler(pClass, genericType, anns, message);
             if (pm != null) {
-                result = pm.fromString(value);
+                return Optional.of(pm);
             }
         }
-        return result;
+        
+        return Optional.empty();
```

Comment:
Minor suggestion - you could replace the if-check and this line with `return Optional.ofNullable(pm);`

Reply:
Surely, less code, thanks @andymc12!